lib/winston/logger.js

```diff
@@ -233,34 +214,28 @@ class Logger extends Transform {
     }
 
     const [meta] = splat;
+    const info = {};
+    this._addDefaultMeta(info);
+    let mergeMessage = false;
     if (typeof meta === 'object' && meta !== null) {
       // Extract tokens, if none available default to empty array to
       // ensure consistancy in expected results
       const tokens = msg && msg.match && msg.match(formatRegExp);
 
       if (!tokens) {
-        const info = Object.assign({}, this.defaultMeta, meta, {
-          [LEVEL]: level,
-          [SPLAT]: splat,
-          level,
-          message: msg
-        });
-
-        if (meta.message) info.message = `${info.message} ${meta.message}`;
-        if (meta.stack) info.stack = meta.stack;
-
-        this.write(info);
-        return this;
+        assignWithGetters(info, meta);
+        mergeMessage = true;
       }
     }
-
-    this.write(Object.assign({}, this.defaultMeta, {
+    assignWithGetters(info, {
       [LEVEL]: level,
       [SPLAT]: splat,
       level,
       message: msg
-    }));
-
+    });
+    if (mergeMessage && meta.message) info.message = `${info.message} ${meta.message}`;
+    if (mergeMessage && meta.stack) info.stack = meta.stack;
```

Comment:
I don't really think this is a good idea, btw - but it keeps consistency with the existing behavior.